Fix/prevent damp power inversion

damosvil opened the pull request:
Set Flag_Rcp_Stop if it rcpulse is less or equal to deadtime,
so damp register is always triggered before power register

Burdalfis reviewed (commented):
Looks good, can test in a few hours if need be.

damosvil commented:
I tested it on my side, but please, when possible could you make it a 5min smoke test?

stylesuxx commented on Bluejay.asm:
```diff
@@ -955,8 +955,16 @@ t1_int_startup_boosted:
+    ; Only set Flag_Rcp_Stop if it is less or equal to deadtime,
+    ; so damp register is always triggered before than power register
+    clr C
+	mov	A, Temp4
```
One space too much ;-)

damosvil replied:
fixed

stylesuxx reviewed (changes requested):
Could you explain the current behavior and consequences of this change? Looking at the code I would assume that RCP stop flag has never been set unless running on a "0" deadtime MCU?

If this is the case are we sure, that whatever is executed in the code path that is now triggered is properly working? Or is it currently being executed in different cases too?

Also you tagged this as "enhancement" - this rather seems like a fix to me. I know, it's only semantics - just wanted to double check.

damosvil commented:
> Could you explain the current behavior and consequences of this change? Looking at the code I would assume that RCP stop flag has never been set unless running on a "0" deadtime MCU?

Yes, previously rcp stop was set only when rcp was 0.

This change will set rc pulse to 0 always it is smaller than deadtime + 1, because this way damp register is set to deadtime and power register is always bigger than damp register.
Remember that when PCA counter overflows and is set to 0 channel output is set to 0, then when counter reaches damp value channel output is inverted and set to 1, and when counter reaches power value channel output is inverted again and is set to 0.
When 0 deadtime rcp stop will be set when pwm is less that 1, because I added 1 unit to DEADTIME.
 
> If this is the case are we sure, that whatever is executed in the code path that is now triggered is properly working? Or is it currently being executed in different cases too?

It sets rcpulse to 0 in all cases

> Also you tagged this as "enhancement" - this rather seems like a fix to me. I know, it's only semantics - just wanted to double check.

Changed to fix

damosvil commented:
I am canceling this pr because pca doesn't work the way I figured out. This pr is doesn't fix any issue
